src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java

```diff
@@ -948,4 +983,173 @@ public static boolean isInSparseFormat(MatrixObject mo) {
+	 * @throws DMLRuntimeException
+	 */
+	public static void vectorScalarMult(ExecutionContext ec, MatrixObject in, double constant, String outputName, GPUEnabledElementwiseOp op) throws DMLRuntimeException {
+		if(isInSparseFormat(in)) {
+			// TODO: Implement sparse vector scalar multiply kernel
+			// throw new DMLRuntimeException("Sparse vector scalar multiply is not supported");
+			((JCudaObject)in.getGPUObject()).sparseToDense();
+		}
+		
+		MatrixObject out = ec.getMatrixObject(outputName);
+	    ec.getDenseMatrixOutputForGPUInstruction(outputName);	// Allocated the dense output matrix
+	    
+		Pointer alpha = pointerTo(constant);
+		if(op == GPUEnabledElementwiseOp.DIVIDE)
+			alpha = pointerTo(1/constant);
+		
+		int incx = 1, incy = 0;
+		int n = (int) in.getNumRows() == 1 ? (int) in.getNumColumns() : (int) in.getNumRows();
+		
+		Pointer A = ((JCudaObject)in.getGPUObject()).jcudaDenseMatrixPtr;
+	    Pointer C = ((JCudaObject)out.getGPUObject()).jcudaDenseMatrixPtr;
+	    
+		JCublas2.cublasDaxpy(cublasHandle, n, alpha, A, incx, C, incy);
+	}
+	
+	/**
+	 * Performs elementwise multiplication or division of input matrix in and constant using cublasDgeam
+	 * 
+	 * @param ec
+	 * @param in
+	 * @param constant
+	 * @param outputName
+	 * @param op
+	 * @param isTransposed
+	 * @throws DMLRuntimeException
+	 */
+	public static void matScalarElementwiseMultDiv(ExecutionContext ec, MatrixObject in, 
+			double constant, String outputName, GPUEnabledElementwiseOp op, boolean isTransposed) throws DMLRuntimeException {
+		if(isInSparseFormat(in)) {
+			// TODO: Implement sparse matrix scalar elementwise  multiply/div kernel
+			// throw new DMLRuntimeException("Sparse matrix scalar elementwise multiply/div is not supported");
+			((JCudaObject)in.getGPUObject()).sparseToDense();
+		}
+		
+		MatrixObject out = ec.getMatrixObject(outputName);
+	    ec.getDenseMatrixOutputForGPUInstruction(outputName);	// Allocated the dense output matrix
+	    
+		Pointer alpha = pointerTo(constant);
+		if(op == GPUEnabledElementwiseOp.DIVIDE)
+			alpha = pointerTo(1/constant);
+		Pointer beta = pointerTo(0.0);
+		
+		int transa = isTransposed ? CUBLAS_OP_T : CUBLAS_OP_N;
+		
+		int m = (int) in.getNumRows();
+		int n = (int) in.getNumColumns();
+		int lda = isTransposed ? n : m; //m;
+		int ldc = m;
+		
+		Pointer A = ((JCudaObject)in.getGPUObject()).jcudaDenseMatrixPtr;
+	    Pointer C = ((JCudaObject)out.getGPUObject()).jcudaDenseMatrixPtr;
+	    
+	    JCublas2.cublasDgeam(cublasHandle, transa, transa, m, n, alpha, A, lda, beta, A, lda, C, ldc);
+	}
+	
+	/**
+	 * Performs elementwise addition or subtraction of two input matrices in1 and in2 using cublasDgeam
+	 * 
+	 * @param ec
+	 * @param in1
+	 * @param in2
+	 * @param outputName
+	 * @param isLeftTransposed
+	 * @param isRightTransposed
+	 * @param isAdd
+	 * @throws DMLRuntimeException
+	 */
+	public static void cellwiseMatMatAddSub(ExecutionContext ec, MatrixObject in1, MatrixObject in2, 
```

Comment:
Match parameters to [cublasDgeam](http://docs.nvidia.com/cuda/cublas/#cublas-lt-t-gt-geam) (transa, transb, A, B...).
Rename to **matMatAdd**. The "element-wise" part is implied.